Dockerfile

```
@@ -15,6 +12,8 @@ RUN --mount=type=cache,target=/root/.cargo/git/db \
     CARGO_REGISTRIES_CRATES_IO_PROTOCOL=sparse \
     cargo fetch
 
+ARG TARGETPLATFORM
+
 RUN xx-apk add --no-cache musl-dev openssl-dev openssl-libs-static
```

If the silent failure is not desirable, maybe we should explicitly check that this argument is defined.

i.e.
```suggestion
ARG TARGETPLATFORM

RUN if [[ -z "$TARGETPLATFORM" ]] ; then echo "argument TARGETPLATFORM was not provided"; exit 1; fi

RUN xx-apk add --no-cache musl-dev openssl-dev openssl-libs-static
```

---

I thought about that, but it seems kind of redundant to have a line checking if a variable is defined right after the line that defines the variable.

I guess it would protect against someone moving the `ARG` in the future, but only if they don't also move the check with it.

A separate CI step that pulls and runs the built image (using qemu) would probably make more sense, but I haven't looked into that yet. It would be a lot easier with GitHub arm runners. In June GitHub said that they will be available end of year, but haven't said anything since.